jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -252,87 +312,58 @@ static void afterClinitInitialization() {
 		StringCoding.encode(new char[1], 0, 1);
 		/*[ENDIF]*/
 
-		/*[IF Sidecar18-SE-OpenJ9 | JAVA_SPEC_VERSION >= 11]*/
+		/*[IF Sidecar18-SE-OpenJ9]*/
 		Properties props = internalGetProperties();
 		/*[IF JAVA_SPEC_VERSION >= 11]*/
 		consoleDefaultCharset = Charset.defaultCharset();
 		/*[IF PLATFORM-mz31|PLATFORM-mz64]*/
 		try {
-			consoleDefaultCharset = Charset.forName(props.getProperty("ibm.system.encoding")); //$NON-NLS-1$
+			consoleDefaultEncoding = props.getProperty("console.encoding"); //$NON-NLS-1$
+			if (consoleDefaultEncoding == null) {
+				consoleDefaultEncoding = props.getProperty("ibm.system.encoding"); //$NON-NLS-1$
+			}
+			consoleDefaultCharset = Charset.forName(consoleDefaultEncoding); //$NON-NLS-1$
```

Comment:
This is `zOS` specific to determine the default charset, wondering if it could go to `Charset.defaultCharset()` as a JCL patch instead?
If so, the variable `consoleDefaultCharset` can be removed, and use `Charset.defaultCharset()` which is a getter after first invocation.

Reply:
This is the console Charset, which on z/OS is different from the Charset.defaultCharset().